### PR TITLE
Emails to s3   add api gateway to s3 bucket

### PR DESCRIPTION
**Pull request by david-pepper**
## What does this change?
Instead of importing emails via an IAM User in AWS, we will use an API gateway. This is to align with current security standards for access to AWS resources.

## How to test

- deploy the stack via cloudformation

**GET 200**

- ensure a test file is present in the s3 bucket
- perform a GET call to the API Gateway
- verify the file contents are returned in a 200 response

**GET 4xx**

- perform a GET call using incorrect credentials to the API Gateway
- verify the file contents are not returned in a 4xx response

**GET 500**

- ensure a test file is not present in the s3 bucket
- perform a GET call to the API Gateway
- verify 500 response is returned

**DELETE 200**

- ensure a test file is present in the s3 bucket
- perform a DELETE call to the API Gateway
- verify the file is deleted

**DELETE 4xx**

- ensure a test file is present in the s3 bucket
- perform a DELETE call using incorrect credentials to the API Gateway
- verify 4xx response is returned and file is not deleted

**DELETE 500**

- ensure a test file is not present in the s3 bucket
- perform a DELETE call to the API Gateway
- verify 500 response is returned

## Notes
I've added a ManagedPolicy to restrict access to the specific s3 bucket and its objects. The ManagedPolicy has been applied to a new IAM Rolem and the API Gateway methods are enabled to assume this role for the relevant GET and DELETE actions on the bucket's objects

### CODE REVIEW

**Review by kelvin-chappell — APPROVED** [review]
This is great to see the cloudformation config to connect from API gateway directly to S3.  I think this is going to be referenced a lot! 💯

File: handlers/sf-emails-to-s3-exporter/cfn.yaml [hunk 1]
```diff
@@ -31,15 +31,34 @@ Resources:
...
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:GetObject",
+                        "s3:DeleteObject"
+                    ],
+                    "Resource": "arn:aws:s3:::emails-from-sf-dev/*"
```
Comment (kelvin-chappell) [inline handlers/sf-emails-to-s3-exporter/cfn.yaml:57]:
```suggestion
                    "Resource": !Sub "arn:aws:s3:::emails-from-sf-${Stage}/*"
```

**Review by kelvin-chappell — COMMENTED** [review]
Looks good!  Just one hardcoded ref to Dev I found.

File: handlers/sf-emails-to-s3-exporter/cfn.yaml (hunk 1 above)
Comment (david-pepper) [reply handlers/sf-emails-to-s3-exporter/cfn.yaml]: Great spot! fixed in this [commit](https://github.com/guardian/support-service-lambdas/pull/1365/commits/db3b1dc964adda11ed0bedced5ff119ceb6c2763) (the !Sub instruction needed to be outside the definition)

**Comment by david-pepper** [comment]
[Add API Gateway for import](https://trello.com/c/GQ08ue6m/2843-add-api-gateway-for-import)

**Pull request merged**